all-construct/all_construct.py

```diff
@@ -1,6 +1,19 @@
 def all_construct(target_word, word_bank):
-  '''
-    Returns an array containing all the ways that the `target` can be constructed
-    by concatenating elements of the `word_bank` array
-  '''
+  if target_word == "":
+        return [[]] 
+    
+  result = []
+    
+  for word in word_bank:
+    if target_word.startswith(word):
+      suffix = target_word[len(word):]  
+      suffix_ways = all_construct(suffix, word_bank)  
+      
+      target_ways = [[word] + way for way in suffix_ways]
+      result.extend(target_ways)
+
+    return result
+
+
+
   return target_word
```

Comment:
this line can be removed, as it returns the target_word. this return is not appropereate for the case that the target word cannot be constructed from the available wordbank.